File: solve/0506_과일장수/postforty/과일 장수.js
function solution(k, m, score) {
  var answer = 0;
  score.sort((a, b) => b - a);
  console.log(score);
  answer = score.reduce((a, b, i) => {
    if (i % m === m - 1) {
      a += b * m;
    }
    return a;
  }, 0);
  return answer;
}

console.log(solution(3, 4, [1, 2, 3, 1, 2, 3, 1]));
console.log(solution(4, 3, [4, 1, 2, 2, 4, 4, 4, 4, 1, 2, 4, 2]));

File: solve/0506_과일장수/postforty/과일 장수.py
def solution(k, m, score):
    # score.sort(reverse=True)
    # answer = sum([score[i] * m for i in range(m-1, len(score), m)])
    answer = (lambda _, m, s: sum(sorted(s)[-m::-m]) * m)(k, m, score)
    return answer

print(solution(3, 4, [1, 2, 3, 1, 2, 3, 1]))
# print(solution(4, 3, [4, 1, 2, 2, 4, 4, 4, 4, 1, 2, 4, 2]))